### PR TITLE
Send payment receipt exactly once

### LINKED ISSUE #2598: Payment receipt email may not have Square receipt URL in some cases
The `payment.created` event does not include the receipt URL, but that's what we're using for triggering a receipt right now, since it's exactly-once.

### PR DESCRIPTION
**Pull request by kberzinch**
Fix #2598

### CODE REVIEW

**Review by JoshOldenburg — APPROVED** [review]
I'm going to trust you on the `Cache::lock` logic.

**Comment by kberzinch** [comment]
Idk if it's 100% correct, the docs are light on details. But i think it's close enough at least.

File: app/Observers/PaymentObserver.php [hunk 1]
```diff
@@ -27,5 +29,22 @@ public function saved(Payment $payment): void
         ) {
             SendDuesPaymentReminder::dispatch($payment->payable->user);
         }
+
+        // this is pretty cursed but i don't have a better idea on guaranteeing exactly one receipt email from
+        // ~four save events
+        Cache::lock('send_payment_receipt_'.$payment->id, 5)->get(static function () use ($payment): void {
```
Comment (evan10s) [inline app/Observers/PaymentObserver.php:35]: The documentation about this function is lacking details and I found myself seeking out answers in the original source code.  The biggest issue is what the 5 means, I would suggest extracting it into a variable and adding a comment that it'll wait up to 5 seconds to obtain the lock ([if I understand correctly](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Cache/Lock.php#L26)).

Comment (evan10s) [inline app/Observers/PaymentObserver.php:35]: Do you not need to catch the `LockTimeoutException` in the event the lock can't be acquired?

Comment (kberzinch) [reply app/Observers/PaymentObserver.php:35]: That was also my understanding. I'll add a comment.

Comment (kberzinch) [reply app/Observers/PaymentObserver.php:35]: Not sure what we would do with it. Just ignore it i guess?

Comment (evan10s) [reply app/Observers/PaymentObserver.php:35]: Ignoring it seems fine

Comment (kberzinch) [reply app/Observers/PaymentObserver.php:35]: After further consideration i don't think this implementation will be a problem in practice, if it shows up in sentry I'll stick it in a try/catch

**Pull request merged**